source/Octopus.Server.Client/OctopusAsyncClient.cs

```diff
@@ -633,9 +633,11 @@ protected virtual async Task<OctopusResponse<TResponseResource>> DispatchRequest
                         return octopusResponse;
                     }
                 }
-                catch (TaskCanceledException)
+                catch (TaskCanceledException exception) when (
+                    exception.CancellationToken != cancellationToken &&
```

Comment:
Should/can this be `is not` instead of `!=`?

Reply:
It can't unfortunately, pattern matching only supports compile time constants as values, which `cancellationToken` is not. IIRC it's being considered for a future C# version though 🙂